src/primitives/createLazyLoadQuery.ts

```diff
@@ -1,49 +1,164 @@
 import type { MaybeAccessor } from "@solid-primitives/utils";
-import { __internal } from "relay-runtime";
-import type {
-	CacheConfig,
-	FetchPolicy,
-	GraphQLTaggedNode,
-	OperationType,
-	RenderPolicy,
-	VariablesOf,
+import RelayRuntime from "relay-runtime";
+import {
+	type CacheConfig,
+	type FetchQueryFetchPolicy,
+	type GraphQLResponse,
+	type GraphQLTaggedNode,
+	Observable,
+	type OperationDescriptor,
+	type OperationType,
+	type VariablesOf,
 } from "relay-runtime";
-import { createMemo } from "solid-js";
-import type { ResourceReturn } from "solid-js";
+import {
+	batch,
+	createComputed,
+	createResource,
+	createSignal,
+	onCleanup,
+} from "solid-js";
+import { createStore, reconcile } from "solid-js/store";
 
 import { useRelayEnvironment } from "../RelayEnvironment";
-import { createLazyLoadQueryNode } from "./createLazyLoadQueryNode";
+import { type DataProxy, makeDataProxy } from "../utils/dataProxy";
 import { createMemoOperationDescriptor } from "./createMemoOperationDescriptor";
 
+type QueryResult<T> = {
+	data: T | undefined;
+	error: unknown;
+	pending: boolean;
+	inFlight: boolean;
+};
+
 export function createLazyLoadQuery<TQuery extends OperationType>(
 	gqlQuery: MaybeAccessor<GraphQLTaggedNode>,
 	variables: MaybeAccessor<VariablesOf<TQuery>>,
 	options?: {
-		fetchKey?: MaybeAccessor<string | number | undefined>;
-		fetchPolicy?: MaybeAccessor<FetchPolicy | undefined>;
+		fetchPolicy?: MaybeAccessor<FetchQueryFetchPolicy | undefined>;
 		networkCacheConfig?: MaybeAccessor<CacheConfig | undefined>;
-		UNSTABLE_renderPolicy?: MaybeAccessor<RenderPolicy | undefined>;
 	},
-): ResourceReturn<TQuery["response"]> {
+): DataProxy<TQuery["response"]> {
 	const environment = useRelayEnvironment();
 
-	const query = createMemoOperationDescriptor(
+	const operation = createMemoOperationDescriptor(
 		gqlQuery,
 		variables,
-		() => access(options?.networkCacheConfig) ?? { force: true },
+		options?.networkCacheConfig,
+	);
+	const [source, setSource] = createSignal<{
+		observable: Observable<GraphQLResponse>;
+		operation: OperationDescriptor;
+	}>();
+	const [serverData, setServerData] = createSignal<TQuery["response"]>();
+
+	const [resource] = createResource(
+		operation,
+		async (operation) => {
+			const observable = RelayRuntime.__internal.fetchQuery(
+				environment,
+				operation,
+			);
+			setSource({ observable, operation });
+			const stream = new ReadableStream<GraphQLResponse>({
+				start(controller) {
+					observable.subscribe({
+						next(response) {
+							controller.enqueue(response);
+						},
+						error(error: unknown) {
+							controller.error(error);
+						},
+						complete() {
+							controller.close();
+						},
+					});
+				},
+			});
+			await observable.toPromise();
+			return stream;
+		},
+		{
+			async onHydrated(operation, { value }) {
+				if (!operation || !value) return;
+
+				const replaySubject = new RelayRuntime.ReplaySubject<GraphQLResponse>();
```

Comment:
ReplaySubject is an observable that retains all of its events so that a future subscriber can get caught up on any events that it missed. I don't think you need that here.